_posts/2020-06-30-hhvm-4.64.markdown

```diff
@@ -0,0 +1,58 @@
+---
+title: "HHVM 4.64"
+layout: post
+author: jjergus
+category: blog
+---
+
+HHVM 4.64 is released! This release marks the end of support for 4.57;
+HHVM 4.58&ndash;4.62 remain supported (4.63 was skipped), as do the 4.32
+and 4.56 LTS releases.
+
+# Breaking Changes
+
+- The INI flag `hhvm.hack_arr_compat_specialization` was removed and its
+  behavior is now enabled by default.
+  This change disables the implicit inter-operability between `varray`s and
+  `darray`s. Instead, `varray`, `darray` and `array` (soon to be removed) are
+  now treated as types in their own right. Specifically:
+  * Using a `varray` where a `darray` is expected (at parameter and return value
```

Comment:
I think it's also worth mentioning property typehint invariance across class hierarchies. Here's what I wrote in the pull request to update the array docs:

If property type enforcement is turned on, property type invariance rules for class hierarchies will apply to `varray` and `darray` and `array`. For example, a child class will not be able to redeclare a property with type `varray<Tv>` if the parent class's property is typed `darray<Tk, Tv>`.

Reply:
Wait, wasn't that already the case? I understand that `$foo->my_darray = varray[]` is now an error under property type enforcement, but didn't we already ban redeclaring properties with incompatible types?

```
class Foo { public ?darray<string, int> $x; }
class FooChild extends Foo {
  public ?varray<int> $x;
}
```